src/http-compute-js/http-outgoing.ts

```diff
@@ -155,7 +155,7 @@ export class WrittenDataBuffer {
 export type HeadersSentEvent = {
   statusCode: number,
   statusMessage: string,
-  headers: Record<string, string>,
+  headers: [header: string, value: string][],
```

Comment:
This is the main change. We can also change this to use `OutgoingHttpHeaders` if that works better!